src/condor_daemon_client/dc_schedd.h

```diff
@@ -405,14 +411,24 @@ class DCSchedd : public Daemon {
 	// returns Q_OK (0) on success
 	// returns one of errors from query_result_type.h on failure (like Q_SCHEDD_COMMUNICATION_ERROR)
 	int queryJobs (int cmd, // QUERY_JOB_ADS or QUERY_JOB_ADS_WITH_AUTH
-		ClassAd & query_ad,
+		const ClassAd & query_ad,
 		// return false to take ownership of the ad, true to allow the ad to be deleted after
 		bool (*process_func)(void*, ClassAd *ad),
 		void * process_func_data,
 		int connect_timeout,
 		CondorError *errstack,
 		ClassAd ** psummary_ad);
 
+#ifdef ENABLE_RAW_JOB_QUERY
+	int queryJobs (int cmd, // QUERY_JOB_ADS or QUERY_JOB_ADS_WITH_AUTH
+		const ClassAd & query_ad,
+		bool (*process_func)(void*, class WireClassadBuffer & wab),
+		void * process_func_data,
```

Comment:
Now that we have C++20 everywhere, strongly consider making this pair of arguments a single `std::function`, instead.  Since you can assign lambdas to `std::function`s, you can bring along as many or as few context arguments as seems appropriate, and in a type-safe way.  (Doubly so since it looks like the only place this function is actually called is already passing a lambda.)

Reply:
A good suggestion, but that would make this a bit less of a drop-in replacement for current users of the schedd query API.   I'd prefer to make that a future enhancement and spend some time looking at how much of a barrier that adds for condor_q etc first.